### PR TITLE
Fix symbolic tensor shape

### LINKED ISSUE #2320: Channel Found None in MaxUnPooling2D
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Colab
- TensorFlow version and how it was installed (source or binary): 2.4.0
- TensorFlow-Addons version and how it was installed (source or binary):
- Python version: 3.7
- Is GPU used? (yes/no):  yes

**Describe the bug**

The channel dimension is None after MaxUnpooling2D layer

**Code to reproduce the issue**

```
def test():
    inputs = tf.keras.layers.Input((None, None, 1))
    x,indices = tf.nn.max_pool_with_argmax(inputs, 2, 2, padding="SAME")
    x = tfa.layers.MaxUnpooling2D()(x,indices)
    x = tf.keras.layers.Conv2D(64, 3, padding="same")(x)
    return tf.keras.Model(inputs, x)
model = test()
model(tf.random.normal([1,128,128,1]))
```

**Other info / logs**

<ipython-input-25-675633950ad9> in test()
      3     x,indices = tf.nn.max_pool_with_argmax(inputs, 2, 2, padding="SAME")
      4     x = tfa.layers.MaxUnpooling2D()(x,indices)
----> 5     x = tf.keras.layers.Conv2D(64, 3, padding="same")(x)
      6     return tf.keras.Model(inputs, x)
      7 model = test()

/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/base_layer.py in __call__(self, *args, **kwargs)
    950     if _in_functional_construction_mode(self, inputs, args, kwargs, input_list):
    951       return self._functional_construction_call(inputs, args, kwargs,
--> 952                                                 input_list)
    953 
    954     # Maintains info about the `Layer.call` stack.

/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/base_layer.py in _functional_construction_call(self, inputs, args, kwargs, input_list)
   1089         # Check input assumptions set after layer building, e.g. input shape.
   1090         outputs = self._keras_tensor_symbolic_call(
-> 1091             inputs, input_masks, args, kwargs)
   1092 
   1093         if outputs is None:

/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/base_layer.py in _keras_tensor_symbolic_call(self, inputs, input_masks, args, kwargs)
    820       return nest.map_structure(keras_tensor.KerasTensor, output_signature)
    821     else:
--> 822       return self._infer_output_signature(inputs, args, kwargs, input_masks)
    823 
    824   def _infer_output_signature(self, inputs, args, kwargs, input_masks):

/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/base_layer.py in _infer_output_signature(self, inputs, args, kwargs, input_masks)
    860           # overridden).
    861           # TODO(kaftan): do we maybe_build here, or have we already done it?
--> 862           self._maybe_build(inputs)
    863           outputs = call_fn(inputs, *args, **kwargs)
    864 

/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/base_layer.py in _maybe_build(self, inputs)
   2708         # operations.
   2709         with tf_utils.maybe_init_scope(self):
-> 2710           self.build(input_shapes)  # pylint:disable=not-callable
   2711       # We must set also ensure that the layer is marked as built, and the build
   2712       # shape is stored since user defined build functions may not be calling

/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/layers/convolutional.py in build(self, input_shape)
    186   def build(self, input_shape):
    187     input_shape = tensor_shape.TensorShape(input_shape)
--> 188     input_channel = self._get_input_channel(input_shape)
    189     if input_channel % self.groups != 0:
    190       raise ValueError(

/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/layers/convolutional.py in _get_input_channel(self, input_shape)
    358     channel_axis = self._get_channel_axis()
    359     if input_shape.dims[channel_axis].value is None:
--> 360       raise ValueError('The channel dimension of the inputs '
    361                        'should be defined. Found `None`.')
    362     return int(input_shape[channel_axis])

ValueError: The channel dimension of the inputs should be defined. Found `None`.

### PR DESCRIPTION
**Pull request by WindQAQ**
# Description

Fixes #2320.

## Type of change

- [x] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [x] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

Passes existing and new tests.

### CODE REVIEW

**Review by thaink — APPROVED** [review]

File: tensorflow_addons/layers/max_unpooling_2d.py [hunk 1]
```diff
@@ -59,6 +59,7 @@ def _max_unpooling_2d(updates, mask, pool_size=(2, 2), strides=(2, 2), padding="
     def func(updates, mask):
         mask = tf.cast(mask, "int32")
         input_shape = tf.shape(updates, out_type="int32")
```
Comment (thaink) [inline tensorflow_addons/layers/max_unpooling_2d.py:61]: Do we need this line after adding line #62?

Comment (WindQAQ) [reply tensorflow_addons/layers/max_unpooling_2d.py:61]: It's still needed to ensure the function work with tensors of partially unknown shapes.

Comment (thaink) [reply tensorflow_addons/layers/max_unpooling_2d.py:61]: acked.

**Review by seanpmorgan — APPROVED** [review]
LGTM as well.

**Pull request merged**